lib/typescript/package.json

```diff
@@ -12,20 +12,15 @@
     "neeva",
     "xyzapi"
   ],
-  "publishConfig": {
-    "registry": "https://npm.pkg.github.com"
-  },
-  "main": "./dist/index.js",
-  "type": "module",
-  "module": "./dist/index.js",
-  "exports": {
-    ".": "./dist/index.js"
-  },
+  "source": "index.ts",
+  "main": "dist/index.js",
+  "type":"module",
+  "types": "dist/index.d.ts",
   "homepage": "https://n.xyz",
-  "typings": "./dist/index.d.ts",
   "scripts": {
-    "update": "mv package.json ./../../ && rm -R ./* && openapi-generator generate -i https://api.n.xyz/api.json --additional-properties=supportsES6=true,typescriptThreePlus=true -g typescript  -o /tmp/typescript && cp -R /tmp/typescript/* ./ && rm -R /tmp/typescript/* && mv -f ./../../package.json ./ && npm install",
-    "build": "tsc",
+    "update": "mv package.json index.ts ./../../ && rm -R ./* && openapi-generator generate -i https://api.n.xyz/api.json --additional-properties=supportsES6=true,typescriptThreePlus=true -g typescript  -o /tmp/typescript && cp -R /tmp/typescript/* ./ && rm -R /tmp/typescript/* && mv -f ./../../package.json ./../../index.ts ./ && npm install",
```

Comment:
This feels brittle. Can you remove the package from the generator code before copying the code to this directory? That way if the command crashes you’re still guaranteed that the files will be cleaned up.

Reply:
Yes that is what this code does from my understanding. the first command moves the package.json and my modified index.ts to the parent directory before doing any removals.

So in the case of a crash package and index.ts will be saved.

Reply:
No if any command prior to && fails, then all subsequent commands will be skipped. If you want the file to be restored, you'd need to use semicolons instead.